runtime/vm/jvminit.c

```diff
@@ -2555,6 +2555,10 @@ VMInitStages(J9JavaVM *vm, IDATA stage, void* reserved)
 			if (OPTION_OK != parseError) {
 				parseErrorOption = VMOPT_XXMAXDIRECTMEMORYSIZEEQUALS;
 				goto _memParseError;
+			} else if (vm->directByteBufferMemoryMax != ~(UDATA)0 && vm->directByteBufferMemoryMax > I_64_MAX) {
+				parseErrorOption = VMOPT_XXMAXDIRECTMEMORYSIZEEQUALS;
+				parseError = OPTION_OUTOFRANGE;
+				goto _memParseError;
 			}
```

Comment:
No need for `else` here. Similar to above, I think we want a different limit for 32-bit platforms.
```c
#if defined(J9VM_ENV_DATA64)
#define DBB_MAX_LIMIT I_64_MAX
#else /* defined(J9VM_ENV_DATA64) */
#define DBB_MAX_LIMIT U_32_MAX
#endif /* defined(J9VM_ENV_DATA64) */
			if ((~(UDATA)0 != vm->directByteBufferMemoryMax)
				&& (vm->directByteBufferMemoryMax > (((UDATA)DBB_MAX_LIMIT) & ~(sizeof(UDATA) - 1)))
			) {
				parseErrorOption = VMOPT_XXMAXDIRECTMEMORYSIZEEQUALS;
				parseError = OPTION_OUTOFRANGE;
				goto _memParseError;
			}
#undef DBB_MAX_LIMIT
```